presto-main/src/main/java/com/facebook/presto/sql/planner/LocalExecutionPlanner.java

```diff
@@ -2324,13 +2333,22 @@ public PhysicalOperation visitTableWriter(TableWriterNode node, LocalExecutionPl
                     statisticsAggregation,
                     getVariableTypes(node.getOutputVariables()),
                     tableCommitContextCodec,
-                    stageExecutionDescriptor.isRecoverableGroupedExecution()
-                            ? ConnectorCommitStrategy.LIFESPAN_COMMIT
-                            : ConnectorCommitStrategy.NO_COMMIT);
+                    getConnectorCommitStrategy());
 
             return new PhysicalOperation(operatorFactory, outputMapping.build(), context, source);
         }
 
+        private ConnectorCommitStrategy getConnectorCommitStrategy()
+        {
+            if (tableWriteCommitRequired) {
+                return TASK_COMMIT;
+            }
+            if (stageExecutionDescriptor.isRecoverableGroupedExecution()) {
```

Comment:
Will there be cases where both `tableWriteCommitRequired` and `stageExecutionDescriptor.isRecoverableGroupedExecution()` is true? And in that case, should we use `TASK_COMMIT` or `LIFESPAN_COMMIT`?

Reply:
@shixuan-fan : Good point. We should use `LIFESPAN_COMMIT` in that case since both `TASK_COMMIT` and `LIFESPAN_COMMIT` are doing "table write commit" (a.k.a. PageSinkCommit) but with different commit granularity (per lifespan vs. per task).

Now I really think Presto's legacy task abstraction is awkward.... All modern SQL system's "task" (F1, Procella) are Presto's "lifespan". Presto's "task" is really a "container" which is managed by a standalone resource managment system.. :)

Reply:
I see. Just to confirm if my following understanding is correct: 

When both `tableWriteCommitRequired` and `stageExecutionDescriptor.isRecoverableGroupedExecution()` is `true`, the only difference between `TASK_COMMIT` and `LIFESPAN_COMMIT` is recoverability, since each task is essentially a lifespan.